testing/trino-tests/src/test/java/io/trino/execution/TestCoordinatorDynamicFiltering.java

```diff
@@ -525,7 +502,26 @@ public ConnectorPageSource createPageSource(
                         List<ColumnHandle> columns,
                         DynamicFilter dynamicFilter)
                 {
-                    return new EmptyPageSource();
+                    return new EmptyPageSource() {
+                        @Override
+                        public CompletableFuture<?> isBlocked()
+                        {
+                            return dynamicFilter.isBlocked();
+                        }
+
+                        @Override
+                        public boolean isFinished()
+                        {
+                            if (!dynamicFilter.isComplete()) {
+                                return false;
+                            }
+
+                            assertEquals(dynamicFilter.getCurrentPredicate(), expectedTableScanDynamicFilter);
+                            assertTrue(dynamicFilter.isBlocked().isDone());
```

Comment:
add a comment why it isn't racy (normally DF might be collected after page source is created)